### PR TITLE
Add lock name on email send command 

### PR DESCRIPTION
**Pull request by mabumusa1**
| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | staging
| Bug fix?                               | no
| New feature?                           | yes
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              | yes
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:
This PR is fixes the failed tests for PR https://github.com/mautic/mautic/pull/8617

> Add lock-name option on 'mautic:emails:send' command. This option allow to have serval email send process running in same time with different lock name.
> 
> A swiftmailler bug is fixed in this PR swiftmailer/swiftmailer#1270 to avoid send the same email twice. This modification should come with next M3 composer update.
> 

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2.

### CODE REVIEW

**Comment by mabumusa1** [comment]
@Dcoutelle do you mind checking this PR, it is your PR with minor fixes

**Comment by mabumusa1** [comment]
@RCheesley Please consider this as well

**Comment by RCheesley** [comment]
@mabumusa1 this will be considered for 3.2 if we can get good tests :) Thanks for picking it up and making the necessary updates, we just haven't had enough time to test.

**Comment by mabumusa1** [comment]
@npracht will be great if you can take a quick look at this

File: app/bundles/EmailBundle/Command/ProcessEmailQueueCommand.php [hunk 1]
```diff
@@ -97,6 +97,24 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
+
+                    if (!$handle = @fopen($file, 'r+')) {
+                        continue;
+                    }
+
+                    if (!flock($handle, LOCK_EX | LOCK_NB)) {
+                        /* This message has just been catched by another process */
+                        continue;
+                    }
+
```
Comment (Dcoutelle) [inline app/bundles/EmailBundle/Command/ProcessEmailQueueCommand.php]: I don't understand why there is row 100 to 108 and row 109 to 117 are dupplicate.

If you locked the handle row 104, why redo a fopen row 109 ?

**Comment by Dcoutelle** [comment]
You removed the ability to add a name to email process from #8617  ( see https://github.com/Webmecanik/mautic/blob/5c9525dacb43c75986353bcbe13ee453535ebd00/app/bundles/EmailBundle/Command/ProcessEmailQueueCommand.php#L65 ) so the goal of this PR is different.

File: app/bundles/EmailBundle/Command/ProcessEmailQueueCommand.php (hunk 1 above)
Comment (Dcoutelle) [reply app/bundles/EmailBundle/Command/ProcessEmailQueueCommand.php]: Also, you never fclose($handle)

Comment (mabumusa1) [reply app/bundles/EmailBundle/Command/ProcessEmailQueueCommand.php]: @Dcoutelle thanks for pointing that out, I pushed the wrong branch when I was rebasing, I made the changes. it should be fine now and match your original PR

**Comment by mabumusa1** [comment]
@Dcoutelle please review again

**Comment by Dcoutelle** [comment]
It look like ok to me, but I don't see the difference with my original PR to male TU work ?

**Comment by mabumusa1** [comment]
> It look like ok to me, but I don't see the difference with my original PR to male TU work ?

This is your commit, it will be attributed to you. but because we did not get a reply on it https://github.com/mautic/mautic/pull/8617#issuecomment-645211261

I duplicated this PR with your code to get the PR moving.

This PR should be attributed to you

**Comment by Dcoutelle** [comment]
I don't really care about PR attribution, I wanted to know why Unit Test are working now 😛

**Review by Dcoutelle — DISMISSED** [review]

**Comment by mabumusa1** [comment]
@npracht this one is ready

**Comment by npracht** [comment]
We need a second reviewer

**Comment by RCheesley** [comment]
@mabumusa1 @Dcoutelle happy to test but there are zero test instructions on either PR. Could somebody enlighten me please?

**Comment by kuzmany** [comment]
@RCheesley This option allow to have serval email send process running in same time with different lock name.

That means If you use queue email processing you can use multiple crons  command in same time:

- `mautic:email:send`
- `mautic:email:send --lock-name=process2`
- `mautic:email:send --lock-name=process3`

etc.
Also need fix in swiftmailer https://github.com/swiftmailer/swiftmailer/pull/1270/files

It's really hard to test it properly. 
You can just set queue mode in emails, and try send emails. 
If you receive then should works. Same like before.

We use it in production

**Pull request merged**